### PR TITLE
Remove filtering of ApiGateway namespace resources

### LINKED ISSUE #1031: [Question] API GW V2 metrics not found with dimensionNameRequirements
Hello,

I have a question regarding the scraping of API GW V2 metrics. I noticed that the supporting PR was released in v0.52.0 for scraping API GW  V2 metrics (HTTP API).

We have this config block in our config.yaml:
```
      # API GW V1 (REST)
      # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html
      - <<: *default_job_config
        type: AWS/ApiGateway
        dimensionNameRequirements:
          - ApiName
          - Stage
        metrics:
          - name: Latency
            statistics: [ Average, Maximum, p95, p99 ]
          - name: Count
            statistics: [ SampleCount, Sum ]
          - name: 4XXError
            statistics: [ Sum, Average ]
          - name: 5XXError
            statistics: [ Sum, Average ]

      # API GW V2 (HTTP)
      # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-metrics.html
      - <<: *default_job_config
        type: AWS/ApiGateway
        dimensionNameRequirements:
          - ApiId
          - Stage
          - Route
        metrics:
          - name: Latency
            statistics: [ Average, Maximum, p95, p99 ]
          - name: Count
            statistics: [ SampleCount, Sum ]
          - name: 4xx
            statistics: [ Sum ]
          - name: 5xx
            statistics: [ Sum ]
```

default_job_config has the values: 
```
default_job_config: &default_job_config
  addCloudwatchTimestamp: true
  roles:
    - <roles>
  regions:
    - eu-west-1
  period: 300
  length: 300
```

We are also using these feature flags: 
```
-enable-feature=aws-sdk-v2,max-dimensions-associator,list-metrics-callback
```

But we are getting these logs when running (multiple times):
```
{"account":"<account>","arn":"<role>","job_type":"AWS/ApiGateway","level":"info","msg":"No metrics data found","region":"eu-west-1","time":"2023-06-22T13:09:42Z"}
```

What are we doing wrong in our config.yaml?

### PR DESCRIPTION
**Pull request by cristiangreco**
In recent yace versions, there's no more need to use the ApiGateway API for fetching gw resources and matching their id in the arn, because this is achieved with the refactored associator and better use of filtering regexes.

Fixes #1031

### CODE REVIEW

**Comment by bogdankatishev** [comment]
Any updates on this?

File: pkg/clients/v2/factory_test.go [hunk 1]
```diff
@@ -398,40 +397,14 @@ func TestCachingFactory_createDMSClient_EnablesFIPS(t *testing.T) {
...
 func TestCachingFactory_createStorageGatewayClient_EnablesFIPS(t *testing.T) {
 	factory, err := NewFactory(logging.NewNopLogger(), jobsCfgWithDefaultRoleAndRegion1, true)
 	require.NoError(t, err)
 
-	client := factory.createAPIGatewayV2Client(factory.clients[defaultRole]["region1"].awsConfig)
+	client := factory.createStorageGatewayClient(factory.clients[defaultRole]["region1"].awsConfig)
 	require.NotNil(t, client)
 
-	options := getOptions[apigatewayv2.Client, apigatewayv2.Options](client)
+	options := getOptions[storagegateway.Client, storagegateway.Options](client)
```
Comment (cristiangreco) [inline pkg/clients/v2/factory_test.go:407]: Drive-by fix.
